### ISSUE TITLE
Please readd full url reference to the suspended tabs

### ISSUE THREAD

**Issue by MarceloPedra** [deanoemcke/thegreatsuspender#158]
The last update is really confusing and of no help :)
Please readd these two things that made things work before:
1) Please readd the option to "Suspend ALL the tabs". Having to suspend current one AND all the others make people lose time when they have to work with 15+ Chrome windows :( 
2) After suspending a tab, show a text with the FULL URL in the resulting tab, instead of only the domain. When you have dozens of pages from the same website open, you want to unsuspend only that one you have to work with in that moment. Otherwise is a mayhem.

I will be happy to donate if you can bring this back. I bet is not that complicated to do :)
Can you? please?
Thank you in advance

**Comment by deanoemcke** [comment]
@MarceloPedra 

in regards to removing the url:
can you give me an example of the different (but similar) pages you might have open? at the moment it will display the page title (not the domain) at the top of the screen. do the pages you have open all have the same page title?
it might help you to know that if you hover over the title it is actually a hyperlink to the original page (so you can see the original url at the bottom of the window when you hover).

in regards to suspend ALL tabs:
it was actually a really popular feature request to change the suspend all tabs button to only suspend OTHER tabs in the window. it sounds like you are trying to suspend all tabs in all windows? i am curious as to why you are doing this? it doesn't sound like something you'd need to do very often? perhaps a solution would be to make a keyboard shortcut which suspends all open tabs in all windows?

interested to hear your thoughts.

**Comment by equalsJeffH** [comment]
Hi deanoemcke,

I strongly second MarceloPedra's item (2) (modulo some details, see below) -- for example, I have many tabs loaded at tools.ietf.org/html/{RFCxxxx,draft-ietf-...}  (or also W3C specs at w3.org) and I use tab-finding extensions such as [a] "Too Many Tabs for Chrome" and [b] "Tab Menu" -- but when my tools.ietf.org tabs are suspended, all that is showing in (most of) those latter tab-finding extensions is "tools.ietf.org" or "w3.org" and I can't discern which tab is which spec (unless the tab is unsuspended at the time). 

as you can see in the screenshot below of [a]'s popup dialog box, most of the tabs are indicating only "tools.ietf.org" and thus I can't determine which spec they have loaded.  An odd thing which I haven't figured out yet is that, for example, "draft-balfanz-..." *is* suspended, yet it is displaying the page title, as are some of the other tabs, tho those others don't appear to be suspended, tho I'm thinking they ought to be suspended because they've been sitting there > 15 min i think without my touching them. 
![screen shot 2015-03-17 at 8 17 22 am](https://cloud.githubusercontent.com/assets/226571/6690508/2b3eeb88-cc7e-11e4-9ce3-0105edc3461c.png)
In terms of which text ought to be displayed as part of a suspended tab's name, I request that it be configurable between the URI or the page title element or both -- this will aid in searching thru tabs using the tab-finding extensions.  it appears that TheGreatSuspender is using page title element? here's another screenshot:
![screen shot 2015-03-17 at 8 28 14 am](https://cloud.githubusercontent.com/assets/226571/6690759/9b4576bc-cc7f-11e4-9031-9918e69d4fe1.png)
Thanks for the nice extension and your work on it -- i generally have a couple hundred tabs lying about and it certainly helps system resources to suspend them.

**Comment by deanoemcke** [comment]
@equalsJeffH hey jeff. thanks for your very detailed response. what you are referring to is a bug. i had hoped i'd fixed it in the latest release. you'll find that if you unsuspend all your tabs and then resuspend them you'll get all the titles.
i'm not sure at this point whether it's only something that is happening during the upgrade process or if it's an ongoing problem. could you please try doing what i suggested (if feasible for you) and then let me know if you continue to find tabs losing the tab title when they are suspended?

**Comment by equalsJeffH** [comment]
@deanoemcke: ok, yes, upon unsuspend all and then resuspend all, the tab titles are available to the tabmanagement extensions -- cool, thanks.   

one new item: I note that the "suspend other tabs" and "unsuspend all tabs" commands apply only to that window and not to all tabs in all windows, which isn't immediately clear.

**Comment by deanoemcke** [comment]
@equalsJeffH thanks for confirming that. could you please let me know if you notice any NEW tabs being suspended without displaying the proper title? would be good to know if there is still a bug out there.

re: 'suspend other tabs'. i made it just the current window to err on the safe side. however the text 'suspend other tabs in this window' was too wordy, so i left out the window part. if you go to the keyboard shortcuts section (in settings) you can actually set keyboard shortcuts for suspend/unsuspend all tabs in all windows. but this functionality is only available through keyboard shortcut.

**Comment by equalsJeffH** [comment]
@deanoemcke: I instantiated a 'new' tab, in a window after doing the unsuspend all and then resuspend all thing, loading a resource not already loaded in any tab in any chrome window.  I let it sit and it eventually was suspended, and yes, it is displaying the proper title. 

re: "the bug" that is behind my initial posting above -- that's on the list to get fixed ?   :)     thanks.

**Comment by deanoemcke** [comment]
@equalsJeffH are you referring to the title not displaying bug? if so, i suspect it may have just been a migration bug in which case it should no longer be an issue so there is no plans to fix it at this point. unless you find it starts happening with newly suspended tabs..

**Comment by equalsJeffH** [comment]
@deanoemcke yes I was referring to the title not displaying bug -- and ok, i'll let u know if there's any further issues, thanks.
